### PR TITLE
Issue #3789 XmlConfiguration Set Property

### PR DESCRIPTION
**Pull request by gregw**
Implement `<Set name="name" property="property"/>`

Signed-off-by: Greg Wilkins <gregw@webtide.com>

### CODE REVIEW

**Comment by gregw** [comment]
@lachlan-roberts can you check if the CI failures of autobahn this branch got are due to something I've done or just flaky?
@janbartel @joakime a review of this would be good.  Plus now we are making a new DTD, is there anything else we should update?

File: jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RuleContainer.java [hunk 1]
```diff
@@ -26,14 +26,15 @@
...
+import org.eclipse.jetty.util.component.Dumpable;
 import org.eclipse.jetty.util.log.Log;
 import org.eclipse.jetty.util.log.Logger;
 
 /**
  * Base container to group rules. Can be extended so that the contained rules
  * will only be applied under certain conditions
  */
-public class RuleContainer extends Rule
+public class RuleContainer extends Rule implements Dumpable
```
Comment (joakime) [inline jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RuleContainer.java]: Seems unrelated

File: jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RuleContainer.java [hunk 2]
```diff
@@ -214,4 +215,10 @@ protected String apply(String target, HttpServletRequest request, HttpServletRes
 
         return target;
     }
+
+    @Override
+    public void dump(Appendable out, String indent) throws IOException
```
Comment (joakime) [inline jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RuleContainer.java]: Ditto, unrelated to this issue.

File: jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RewriteHandler.java [hunk 3]
```diff
@@ -178,6 +178,7 @@
     public RewriteHandler()
     {
         _rules = new RuleContainer();
+        addBean(_rules);
```
Comment (joakime) [inline jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RewriteHandler.java]: Unrelated change.

File: jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RewriteHandler.java [hunk 4]
```diff
@@ -207,6 +208,7 @@ public void setRules(Rule[] rules)
      */
     public void setRuleContainer(RuleContainer rules)
     {
+        updateBean(_rules, rules);
```
Comment (joakime) [inline jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RewriteHandler.java]: Unrelated change.

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java [hunk 5]
```diff
@@ -153,7 +153,7 @@
 {
     public static final String GZIP = "gzip";
     public static final String DEFLATE = "deflate";
-    public static final int DEFAULT_MIN_GZIP_SIZE=16;
+    public static final int DEFAULT_MIN_GZIP_SIZE=2048;
```
Comment (joakime) [inline jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java]: Seems unrelated.

File: jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DispatcherTest.java [hunk 6]
```diff
@@ -163,7 +164,7 @@ public void testForwardWithParam() throws Exception
     @Test
     public void testForwardWithBadParams() throws Exception
     {
-        try(StacklessLogging nostack = new StacklessLogging(ServletHandler.class))
+        try(StacklessLogging nostack = new StacklessLogging(HttpChannel.class))
```
Comment (joakime) [inline jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DispatcherTest.java]: Unrelated.

File: jetty-servlets/src/test/java/org/eclipse/jetty/servlets/IncludedGzipTest.java [hunk 7]
```diff
@@ -90,6 +90,7 @@ public void setUp() throws Exception
         tester.getContext().addServlet(org.eclipse.jetty.servlet.DefaultServlet.class, "/");
         
         GzipHandler gzipHandler = new GzipHandler();
+        gzipHandler.setMinGzipSize(16);
```
Comment (joakime) [inline jetty-servlets/src/test/java/org/eclipse/jetty/servlets/IncludedGzipTest.java]: Unrelated.

File: jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipDefaultTest.java [hunk 8]
```diff
@@ -469,6 +469,7 @@ public void testGzipCompressedByContentTypeWithEncoding() throws Exception
     {
         GzipTester tester = new GzipTester(testingdir.getEmptyPathDir(),compressionType);
         tester.setContentServlet(HttpContentTypeWithEncoding.class);
+        tester.getGzipHandler().setMinGzipSize(16);
```
Comment (joakime) [inline jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipDefaultTest.java]: Unrelated.

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java (hunk 5 above)
Comment (gregw) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java]: This one is related.  We had two different defaults, one in XML and one in code.  In this case I decided to make the default just in the code and the change was needed to pass some tests.

File: jetty-servlets/src/test/java/org/eclipse/jetty/servlets/IncludedGzipTest.java (hunk 7 above)
Comment (gregw) [reply jetty-servlets/src/test/java/org/eclipse/jetty/servlets/IncludedGzipTest.java]: This is related to the default changes.

File: jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipDefaultTest.java (hunk 8 above)
Comment (gregw) [reply jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipDefaultTest.java]: ditto... is related

File: jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RuleContainer.java (hunk 1 above)
Comment (gregw) [reply jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/RuleContainer.java]: yep - was needed to help debug these changes, but is ultimately unrelated.   I will move it to a PR against 9.4 as it is useful there as well.

**Comment by gregw** [comment]
See #3800 for unrelated changes
